stingray/bexvar.py

```diff
@@ -0,0 +1,285 @@
+        obtained for each bin.
+
+    rate_conversion : iterable, `:class:numpy.array` or `:class:List` of floats
+        A list or array of rate conversion constants. Obtained by multipllying
+        fractional exposer with time interval for each bin.
+
+    density_gp : int
+        A number specifying density of grid points in the output ``log_src_crs_grid``.
+
+    Returns
+    -------
+    log_src_crs_grid : iterable, `:class:numpy.array` of floats
+        An array (grid) of log(Source count rates).
+    """
+
+    # lowest count rate
+    a = scipy.special.gammaincinv(src_counts + 1, 0.001) / rate_conversion
+    # highest background count rate
+    b = scipy.special.gammaincinv(bkg_counts + 1, 0.999) / (rate_conversion * bkg_area)
+
+    mindiff = min(a - b)
+    if mindiff > 0: # minimum background-subtracted rate is positive
+        m0 = np.log10(mindiff)
+    else: # minimum background-subtracted rate is negative (more background than source)
+        m0 = -1
+    # highest count rate (including background)
+    c = scipy.special.gammaincinv(src_counts + 1, 0.999) / rate_conversion
+    m1 = np.log10(c.max())
+
+    # add a bit of padding to the bottom and top
+    lo = m0 - 0.05 * (m1 - m0)
+    hi = m1 + 0.05 * (m1 - m0)
+    span = hi - lo
+    if lo < -1:
+        log_src_crs_grid = np.linspace(-1.0, hi, int(np.ceil(density_gp * (hi + 1.0))))
+    else:
+        log_src_crs_grid = np.linspace(lo, hi, int(np.ceil(density_gp * 1.05 * span)))
+
+    return log_src_crs_grid
+
+
+def _estimate_source_cr_marginalised(log_src_crs_grid, src_counts, bkg_counts, bkg_area, \
+        rate_conversion):
+    """
+    Compute the PDF at positions in log(source count rate)s grid ``log_src_crs_grid``
+    for observing ``src_counts`` counts in the source region of size src_area,
+    and ``bkg_counts`` counts in the background region of size ``bkg_area``.
+
+    Parameters
+    ----------
+    log_src_crs_grid : iterable, `:class:numpy.array` of floats
+        An array (grid) of log(Source count rates).
+    src_counts : float
+        Source region counts in one of the bin.
+    bkg_counts : float
+        Background region counts in the bin.
+    bkg_area : float
+        Background region area in the bin.
+    rate_conversion : float
+        Rate conversion constant for the bin. Used to convert counts to count rate.
+
+    Returns
+    -------
+    weights: float
+    """
+    # background counts give background count rates deterministically
+    N = 1000
+    u = np.linspace(0, 1, N)[1:-1]
+    bkg_cr = scipy.special.gammaincinv(bkg_counts + 1, u) / bkg_area
+
+    def prob(log_src_cr):
+        src_cr = 10**log_src_cr * rate_conversion
+        like = scipy.stats.poisson.pmf(src_counts, src_cr + bkg_cr).mean()
+        return like
+
+    weights = np.array([prob(log_src_cr) for log_src_cr in log_src_crs_grid])
+
+    if weights.sum() <= 0:
+        print("WARNING: Weight problem! sum is" , weights.sum(), \

```

Comment:
We use the `warning` module for this. Also, might be worth adding a test for this case, using `pytest.warns` to check that the warning is raised if this condition is produced.

Reply:
I have added the warning using the `warning` module and the[ test](https://github.com/mihirtripathi97/stingray/blob/GSoC-22/stingray/tests/test_bexvar.py#:~:text=def%20test_weights_sum_warning(self)%3A).